scripts/qa_crowbarsetup.sh

```diff
@@ -1475,6 +1475,41 @@ function onadmin_repocleanup()
     zypper mr -d sp3sdk
 }
 
+# replace zypper repos from the image with user-specified ones
+# because clouddata might not be reachable from where this runs
+function onadmin_reporeplace()
+{
+    # Delete all repos except PTF repo, because this could
+    # be called after the addupdaterepo step.
+    zypper lr -e - | sed -n '/^name=/ {s///; /ptf/! p}' | \
+        xargs -r zypper rr
+
+    # restore needed repos depending on localreposdir_target
+    if [ -n "${localreposdir_target}" ]; then
+        mount_localreposdir_target
+    else
+        # restore repos from $clouddata
```

Comment:
I know this is outside the scope of this PR, and you probably didn't even write this code (maybe I did!) but moving code always provides a good opportunity to refactor it ...

If you split this into a separate `add_clouddata_repos` then

- the code would be self-documenting
- it would make this function shorter and easier to read
- it would reduce the amount of nested indentation

Comments are often a bad smell, e.g. see

- https://sourcemaking.com/refactoring/smells/comments
- http://programmers.stackexchange.com/questions/1/comments-are-a-code-smell
- http://blogs.agilefaqs.com/2013/03/07/inline-comments-in-the-code-is-a-smell-but-document-the-why/

Having said all that, it would be fine to fix that in a follow-up PR.